core/src/main/java/org/apache/iceberg/SnapshotSummary.java

```diff
@@ -160,27 +177,154 @@ public void merge(SnapshotSummary.Builder builder) {
       // copy custom summary properties
       builder.putAll(properties);
 
+      metrics.addTo(builder);
+      setIf(deletedDuplicateFiles > 0, builder, DELETED_DUPLICATE_FILES, deletedDuplicateFiles);
+      Set<String> changedPartitions = partitionMetrics.keySet();
+      setIf(trustPartitionMetrics, builder, CHANGED_PARTITION_COUNT_PROP, changedPartitions.size());
+
+      if (trustPartitionMetrics && changedPartitions.size() <= maxChangedPartitionsForSummaries) {
+        setIf(changedPartitions.size() > 0, builder, PARTITION_SUMMARY_PROP, "true");
+        for (String key : changedPartitions) {
+          setIf(key != null, builder, CHANGED_PARTITION_PREFIX + key, partitionSummary(partitionMetrics.get(key)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.